server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/DatafileManager.java

```diff
@@ -506,6 +518,7 @@ Optional<StoredTabletFile> bringMajorCompactionOnline(Set<StoredTabletFile> oldD
           datafileSizes.put(newFile.orElseThrow(), dfv);
           // could be used by a follow on compaction in a multipass compaction
         }
+        tabletLog.compacted(oldDatafiles, newFile);
```

Comment:
It seems like after dataFileSizes and tabletLog are updated that we would always expect them to be equal, so could check that in all the places they are updated.

```suggestion
        tabletLog.compacted(oldDatafiles, newFile);
        if(!tabletLog.getExpectedFiles().equals(datafileSizes.keySet()){
            // TODO log something? these are expected to be equal
        }
```